modules/top-level/output.nix

```diff
@@ -285,12 +285,19 @@ in
         '';
       };
 
-      extraConfigLuaPre = lib.mkIf config.wrapRc ''
-        -- Ignore the user lua configuration
-        vim.opt.runtimepath:remove(vim.fn.stdpath('config'))              -- ~/.config/nvim
-        vim.opt.runtimepath:remove(vim.fn.stdpath('config') .. "/after")  -- ~/.config/nvim/after
-        vim.opt.runtimepath:remove(vim.fn.stdpath('data') .. "/site")     -- ~/.local/share/nvim/site
-      '';
+      extraConfigLuaPre = lib.mkOrder 1 (
+        # Add a global table at start of init
+        ''
+          local M = {}
```

Comment:
Maybe have a lua comment instead of a nix comment, to add context when reading `nixvim-print-init`?

```lua
-- Nixvim's internal module
local M = {}
```

Not sure what we'd actually want to say though. Normally comments should explain _why_ we do something, but comments in the final init.lua don't need to do that as they are more like documentation and headers... :thinking:

Reply:
Also cc @nix-community/nixvim: are we happy with this being `M` or should it be something like `_M` to indicate internal/private scope?

Reply:
Maybe we shoud do something like `_M`, as it may conflict with user defined variables

Reply:
Yeah, that's fair. I don't want to break another user's config if they chose to follow the lua standard for a module table, as well.